### PR TITLE
Update typingStatus to the same format when the request is coming from e.com

### PR DESCRIPTION
**Pull request by stitesExpensify**
### Details
We built our chat 'isTyping' indicators on e.com significantly before we did for e.cash, and we implemented them differently. On e.com we publish with pusher and the data is in this format: `{userLogin: 'Concierge'}`
On e.cash we publish with pusher and the data is in this format: `{'bstites@expensify.com': true}`

That meant that when we got [here in e.cash](https://github.com/Expensify/Expensify.cash/blob/1acc60f09d90ad6bcf8c85212820901323ee91da/src/libs/actions/Report.js#L342) we were setting the key as `userLogin`. 

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/149072#event-4165894115

### Tests
1. Open multiple chrome windows, one with e.cash, one with e.com logged into concierge, and one incognito of e.cash
2. Send a message to concierge from one of the e.cash windows
3. In concierge get the chat you sent, and start typing a reply
4. Make sure you can see `Concierge is typing` in the e.cash window
5. Send a message from one e.cash to the other
6. Load up the chat in both windows
7. Type on each one and make sure it says `<email1> is typing`

I only tested on web and desktop because this change didn't change any UI elements, only logic. 

### Tested On

- [x] Web
- [x] Desktop

### CODE REVIEW

**Comment by stitesExpensify** [comment]
I'm not completely sure how I feel about this change. On one hand, I think it may be better to completely re-write the e.com side to be compatible with e.cash rather than accounting for it here, on the other hand it seems like a lot of work updating UI logic on the e.com side when it will go away eventually. Thoughts (whoever reviews this)?

File: src/libs/actions/Report.js [hunk 1]
```diff
@@ -324,15 +324,23 @@ function subscribeToReportTypingEvents(reportID) {
...
         if (!login) {
             return;
         }
 
+        // If the user is typing on expensify.com (like concierge) the format will actually be [userLogin: Concierge]
+        // We need to make sure to handle that case or it will say "userLogin is typing..."
+        if (login === 'userLogin') {
+            login = typingStatus.userLogin;
+            userObject = {[typingStatus.userLogin]: true};
+        }
```
Comment (marcaaron) [inline src/libs/actions/Report.js:338]: If we are going in this direction... it feels like it should be some kind of utility method to me that takes the `typingStatus` object and converts it over to the correct format something like...

`getNormalizedTypingStatus()`

that way we can clearly see the different shapes this data might take and how we handle it differently and why.

**Review by marcaaron — COMMENTED** [review]
I sort of agree that e.com should be re-written to match this style, but it doesn't feel like a priority for now. Reports aren't really viewable in `Expensify.cash` yet. If/when they are that will increase the probability of you chatting with someone who is viewing a report on `e.com` but there were conversations about ripping that feature out of web and mobile entirely. So in light of that, I think it's ok to do this for now.

File: src/libs/actions/Report.js (hunk 1 above)
Comment (stitesExpensify) [reply src/libs/actions/Report.js:338]: Love it. I'll hop on it in the morning

Comment (stitesExpensify) [reply src/libs/actions/Report.js]: Updated. Thanks for the suggestion

**Comment by stitesExpensify** [comment]
Updated

**Review by marcaaron — COMMENTED** [review]
Yep, awesome looks great!

**Review by Jag96 — APPROVED** [review]
LGTM, tested on mobile as well

**Pull request merged**